server/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -211,15 +211,42 @@ public boolean insert(InsertPlan insertPlan) throws ProcessorException {
     try {
       String[] measurementList = insertPlan.getMeasurements();
       String deviceId = insertPlan.getDeviceId();
-      MNode node = mManager.getNodeByDeviceIdFromCache(deviceId);
       String[] values = insertPlan.getValues();
       TSDataType[] dataTypes = new TSDataType[measurementList.length];
+      IoTDBConfig conf = IoTDBDescriptor.getInstance().getConfig();
+
+      try {
+        mManager.checkDeviceId(deviceId);
+      } catch (StorageGroupException e1) {
+        if (!conf.isAutoCreateSchemaEnable()) {
+          throw new ProcessorException(e1);
+        } else {
+          try {
+            // set storage group by auto level
+            setStorageGroupToMTreeByAutoLevel(deviceId);
+            // create timeserires
+            addPathListToMTree(deviceId, measurementList, values);
+          } catch (ProcessorException | PathErrorException e2) {
+            throw new ProcessorException(e2);
+          }
+        }
+      } catch (PathErrorException e2) {
+        if (!conf.isAutoCreateSchemaEnable()) {
+          throw new ProcessorException(e2);
+        } else {
+          addPathListToMTree(deviceId, measurementList, values);
```

Comment:
I think you can add a flag here and if you have called `addPathListToMTree()`, you can skip the next check since you have already created the time series, there is no chance they do not exist.